plugins/nexus-repository-manager/src/api/nexus-repository-manager-api-client/nexus-repository-manager-api-client.ts

```diff
@@ -96,49 +139,85 @@ export class NexusRepositoryManagerApiClient
     return await SearchService.search(query);
   }
 
-  private async fetcher(url: string, additionalHeaders: HeadersInit = {}) {
+  private async fetcher(
+    url: string,
+    additionalHeaders: HeadersInit = {},
+    method: string = 'GET',
+  ) {
     const { token: idToken } = await this.identityApi.getCredentials();
 
     const headers = new Headers(additionalHeaders);
-    if (!headers.has('Accept')) {
-      headers.set('Accept', 'application/json');
-    }
 
     if (idToken) {
       headers.set('Authorization', `Bearer ${idToken}`);
     }
 
-    const response = await fetch(url, { headers });
+    const response = await fetch(url, { headers, method });
     if (!response.ok) {
       throw new Error(
-        `failed to fetch data, status ${response.status}: ${response.statusText}`,
+        `failed to fetch data, status ${response.status}: ${
+          response.statusText
+        } ${await response.text()}`,
       );
     }
-    return await response.json();
+    return await response;
   }
 
-  private async getRawAsset(url?: string, additionalHeaders: HeadersInit = {}) {
-    const proxyUrl = await this.getBaseUrl();
-
-    if (!url) {
-      return null;
+  /**
+   * Use HEAD requests to get the size of each asset we care about, as nexus
+   * doesn't return that information in the search API.
+   * Only supports maven for now.
+   */
+  private async addFileSizes(component: ComponentXO): Promise<ComponentXO> {
```

Comment:
> Are there any other circumstances where it wouldn't be a `DockerManifestSchema`?

I haven't thought too far ahead, to be honest. But there might be scenarios where you'd want to fetch the `pom.xml` or `package.json` and do something with it... I will rename to `DockerManifest` and if those scenarios come up, it can be changed back.